### ISSUE TITLE
[Bug]: Angular 16 + windows 10 + nodejs 18.15.0

### ISSUE THREAD

**Issue by sevaru** [thymikee/jest-preset-angular#2091]
### Version

13.1.0

### Steps to reproduce

1) Clone repo https://github.com/sevaru/ng-16-jest-preset-angular
2) npm i 
3) npm test
4) error

### Expected behavior

I expect tests to run

### Actual behavior

Error: Jest: Got error running globalSetup - C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\global-setup.js, reason: Cannot find module '@angular\compiler-cli'
Require stack:
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\utils\ngcc-jest-processor.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\config\global-setup.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\global-setup.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-util\build\requireOrImportModule.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-util\build\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-config\build\getCacheDirectory.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-config\build\Defaults.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-config\build\normalize.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-config\build\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-cli\build\init\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-cli\build\run.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest-cli\build\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\jest\build\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular-builders\jest\dist\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular-devkit\architect\node\node-modules-architect-host.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular-devkit\architect\node\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\src\command-builder\architect-base-command-module.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\src\command-builder\architect-command-module.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\src\commands\test\cli.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\src\commands\command-config.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\src\command-builder\command-runner.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\lib\cli\index.js
- C:\repos\ng-16-jest-preset-angular\node_modules\@angular\cli\lib\init.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Function.resolve (node:internal/modules/cjs/helpers:116:19)
    at findAngularCompilerCliVersion (C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\utils\ngcc-jest-processor.js:27:26)
    at runNgccJestProcessor (C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\utils\ngcc-jest-processor.js:61:38)
    at C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\config\global-setup.js:18:56
    at Generator.next (<anonymous>)
    at C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\config\global-setup.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\config\global-setup.js:4:12)
    at globalSetup (C:\repos\ng-16-jest-preset-angular\node_modules\jest-preset-angular\build\config\global-setup.js:14:27)

### Additional context

I've used "@angular-builders/jest": "15.0.0" and install it with --force (cause of peerDeps do not allow angular 16)

I've noticed that problem occurs on the line 
```
const ANGULAR_COMPILER_CLI_PKG_NAME = `@angular${path_1.default.sep}compiler-cli`;
```
in jest-preset-angular/build/utils/ngcc-jest-processor.js

and on windows for some reason that code resolves to @angular\compiler-cli and require.resolve returns MODULE_NOT_FOUND
while for @angular/compiler-cli it could find module

If I understand correctly require.resolve accepts module path not a file path and it should have separator '/' not '\'

Than I search thought source and find out that source code do not have path.sep in it 
```
const ANGULAR_COMPILER_CLI_PKG_NAME = '@angular/compiler-cli';
```
https://github.com/thymikee/jest-preset-angular/blob/128b76c9ae7eaa3df86e59ad592427eac79034f8/src/utils/ngcc-jest-processor.ts#LL8C52-L8C52

Then I've build repo myself and in build folder there are still no path.sep.

So the question is what tool postprocess code in build directory before publish and how can we fix it ?

### Environment

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    jest: 29.5.0 => 29.5.0
```

**Comment by ahnpnl** [comment]
This was already fixed in https://github.com/thymikee/jest-preset-angular/commit/b6b391155204120f31dc5538fee554c93b10d001 but not released yet. With angular 16 you don’t need `global-setup` anymore so workaround for now is you can remove `global-setup` from your Jest config.

**Comment by sevaru** [comment]
Thanks 👍 
I've tried experimental support in angular 16 
But did not figure out how to properly pass TextDecoder, TextEncoder.

And decided to wait and use jest-preset-angular in the meantime.
So I'll wait for the release for now I quess

**Comment by ahnpnl** [comment]
About `TextDecoder`, you can check the discussion here https://github.com/thymikee/jest-preset-angular/issues/1914

**Comment by ahnpnl** [comment]
closed as fixed in release **13.1.1** `TextEncoder` is not included
